From 35ae7c085b7076c0e5825462514f38490a540514 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 26 Jun 2018 08:44:30 +0800
Subject: [PATCH 4092/5242] MLK-18684 ARM64: dts: freescale: imx8qxp: correct
 thermal zone number

commit  a6ccf5d3d39e1452975e83878720177f6770b5e4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Commit (dfe39e3 MLK-18673 ARM64: dts: freescale:
imx8dx: disable DRC thermal zone) removes DRC thermal
zone but did NOT fix the thermal zone number and
index, it causes thermal driver probe fail, this
patch corrects them.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi     |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 8f4aca8..48563a6a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -1356,7 +1356,7 @@
 	tsens: thermal-sensor {
 		compatible = "nxp,imx8qxp-sc-tsens";
 		/* number of the temp sensor on the chip */
-		tsens-num = <2>;
+		tsens-num = <1>;
 		#thermal-sensor-cells = <1>;
 	};
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 1fd07c1..9d73d5c 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -1135,14 +1135,14 @@
 };
 
 &tsens {
-	tsens-num = <3>;
+	tsens-num = <2>;
 };
 
 &thermal_zones {
 	pmic-thermal0 {
 		polling-delay-passive = <250>;
 		polling-delay = <2000>;
-		thermal-sensors = <&tsens 2>;
+		thermal-sensors = <&tsens 1>;
 		trips {
 			pmic_alert0: trip0 {
 				temperature = <80000>;
-- 
1.7.9.5

